### ISSUE TITLE
MethodMutator Match

### ISSUE THREAD

**Issue by scelestino** [stryker-mutator/stryker4s#70]
### Summary

There is an issue with the way that MethodMutator match a Tree. For example, it is looking for a ```Term.Name("filter")``` instead of ```Term.Apply(Term.Select(_, Term.Name("filter")), _)```.

That is a problem because the following code try to mutate to filterNot.

```obj.filter // filter is a field ```
```val filter = 123 // filter is a variable```
```import foo.bar.filter.X // filter is a package```

The problem starts in the object PartialStatement that removes useful trees that can be used to get a better match.

I detected this issue trying to run stryker with a project that use ning, and I have an import ```import com.ning.http.client.filter.{FilterContext, RequestFilter}```

A possible workaround could be keep away import trees from the ```Transformer```.

```
    private class OnceTransformer(liftedFn: Tree => Option[Tree]) extends Transformer {
      override def apply(tree: Tree): Tree =
        tree match {
          case importer @ Importer(_, _) => importer
          case otherwise => liftedFn(otherwise).getOrElse(super.apply(otherwise))
        }
    }
```

But I think the best way is changing the MethodMutator match in a way that reflect a method call.

### Stryker config

```
stryker4s {
  files: [
    "src/main/scala/**/*.scala"
  ]
   base-dir: "/home/scelestino/dev/project"
   log-level: INFO
   test-runner: {
    type: commandrunner
    command: "sbt"
    args: "test"
  }
   reporters: ["console"]
}
```

### Stryker4s environment

```
stryker4s 0.1.0-SNAPSHOT
```
git@github.com:stryker-mutator/stryker4s.git master

### Your Environment

``` Scala version 2.12.6 ```
``` Build tool & version SBT 1.1.6 ```
``` Operating System  Ubuntu 17.10 ```

**Comment by hugo-vrijswijk** [comment]
Thanks for your report! This is definitely something we should fix, and I agree the second option is the way to go. This probably applies to some other mutator categories, too, so it should be fixed there as well. I'm fairly busy this weekend, but I will see if I can give it a go sometime next week (maybe tomorrow). If you're willing to have a look, that'd be great too, of course ;)

**Comment by scelestino** [comment]
Ok, I'm going to give it a look tonight

**Comment by scelestino** [comment]
I [fix](https://github.com/scelestino/stryker4s/tree/Fix-Issue%2370-MethodMutator-Match) this issue, just give me some time to add and fix some broken tests. (I think I found another bug)

**Comment by hugo-vrijswijk** [comment]
Fixed in #75
